utils/utils.go

```diff
@@ -394,22 +397,53 @@ func GetChecksum(_ context.Context, fileName string) (checksum string, err error
 	return hex.EncodeToString(hasher.Sum(nil)), nil
 }
 
-// GetInstalledPackages only support for linux
+// GetInstalledPackages returns a map which contains install packages
 func GetInstalledPackages(ctx context.Context) map[string]bool {
 	m := make(map[string]bool)
-	cmd := exec.Command("dpkg-query", "-f", "${binary:Package} ", "-W")
-	stdout, err := cmd.Output()
-	if err != nil {
-		log.WithContext(ctx).Errorf("failed to execute cmd, err: %s", err)
-		return m
-	}
+	switch GetOS() {
+	case constants.Linux:
+		cmd := exec.Command("dpkg-query", "-f", "${binary:Package} ", "-W")
+		stdout, err := cmd.Output()
+		if err != nil {
+			log.WithContext(ctx).Errorf("failed to execute cmd, err: %s", err)
+			return m
+		}
 
-	packages := strings.Split(string(stdout), " ")
-	for _, p := range packages {
-		tokens := strings.Split(p, ":")
-		if tokens[0] != "" {
-			m[tokens[0]] = true
+		packages := strings.Split(string(stdout), " ")
+		for _, p := range packages {
+			tokens := strings.Split(p, ":")
+			if tokens[0] != "" {
+				m[tokens[0]] = true
+			}
+		}
+	case constants.Mac:
+		paths := os.Getenv("PATH")
+		for _, path := range strings.Split(paths, ":") {
+			files, err := ioutil.ReadDir(path)
+			if err != nil {
+				log.WithContext(ctx).Errorf("failed to read dir, err: %s", err)
+				continue
+			}
+
+			for _, file := range files {
+				m[file.Name()] = true
+			}
 		}
 	}
 	return m
 }
+
+// GetServiceConfig returns service configuration
+func GetServiceConfig(name string, format string, value interface{}) (string, error) {
+	temp, err := template.New(name).Parse(format)
```

Comment:
Is this `html/template` or `texte/template` ?

Reply:
ahh. should `text/template`. will fix in next commit

Reply:
i think we should use `text/template`